packages/happy-dom/src/nodes/element/Element.ts

```diff
@@ -255,7 +255,23 @@ export default class Element extends Node implements IElement {
 	 * @returns Attributes.
 	 */
 	public get attributes(): INamedNodeMap {
-		return Object.assign(new NamedNodeMap(this), Object.values(this._attributes), this._attributes);
+		return new Proxy(new NamedNodeMap(this), {
+			get: (target, name) => {
+				if (name in target && typeof target[name] === 'function') {
+					return (...args) => target[name](...args);
+				}
+
+				if (typeof name === 'symbol') {
+					return target[name];
+				}
+
+				if (typeof name === 'string' && /^\d+$/.test(name)) {
+					return Object.values(this._attributes)[parseInt(name, 10)];
+				}
+
+				return this._attributes[name] || target[name];
+			}
+		});
```

Comment:
I'm curious where this implementation came from? It seems like you are referencing values from `this._attributes` instead of `target` arbitrarily, but I'm sure I am missing something.

Reply:
Side note, what should happen if there are clashes between property names between `this._attributes` and `Object.values(this._attributes)`? The previous implementation made the properties from `this._attributes` take greater precedence. I'm not sure this really matters though.

Reply:
See https://github.com/capricorn86/happy-dom/pull/876#discussion_r1177014366